src/H5T.c

```diff
@@ -2704,6 +2704,8 @@ H5T__unregister(H5T_pers_t pers, const char *name, H5T_t *src, H5T_t *dst, H5T_c
                 continue;
             if (dst && dst->shared->type != soft->dst)
                 continue;
+            if (owned_vol_obj)
```

Comment:
We don't want to remove the vlen conversion from the soft conversions table here, so always continue on if the `owned_vol_obj` parameter is specified. Ideally, the logic for removing the soft conversion functions would be separate from removing the cached conversion pathway table entries, but this seems minimally invasive.

Reply:
The entire for loop can be skipped if owned_vol_obj is non-NULL.  (So you could add that check to the surrounding if statement, which will reduce the performance impact a little)